src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsTaskFactory.java

```diff
@@ -63,30 +68,35 @@ private void generateForJava(Project project, Action<? super SpotBugsTask> confi
 
   private void generateForAndroid(
       Project project, Action<? super SpotBugsTask> configurationAction) {
-    project
-        .getPlugins()
-        .withId(
-            "com.android.application",
-            plugin ->
-                project
-                    .getTasks()
-                    .withType(AndroidJavaCompile.class)
-                    .all(
-                        task -> {
-                          String name = GUtil.toLowerCamelCase("spotbugs " + task.getVariantName());
-                          log.debug("Creating SpotBugsTask for {}", task);
-                          project
-                              .getTasks()
-                              .register(
-                                  name,
-                                  SpotBugsTask.class,
-                                  spotbugsTask -> {
-                                    spotbugsTask.setSourceDirs(task.getSource());
-                                    spotbugsTask.setClassDirs(
-                                        task.getOutputDirectory().getAsFileTree());
-                                    spotbugsTask.setAuxClassPaths(task.getClasspath());
-                                    configurationAction.execute(spotbugsTask);
-                                  });
-                        }));
+    if (project.getPlugins().hasPlugin("com.android.base")) {
+
+      final BaseExtension baseExtension = project.getExtensions().getByType(BaseExtension.class);
+      DomainObjectSet<? extends BaseVariant> variants;
+      if (baseExtension instanceof AppExtension) {
+        variants = ((AppExtension) baseExtension).getApplicationVariants();
+      } else if (baseExtension instanceof LibraryExtension) {
+        variants = ((LibraryExtension) baseExtension).getLibraryVariants();
+      } else {
+        throw new RuntimeException("Unrecognized Android extension " + baseExtension);
```

Comment:
Better to throw an instance of [`GradleException`](https://docs.gradle.org/current/javadoc/org/gradle/api/GradleException.html) or its subclasses, but I cannot find suitable class at this moment.

Reply:
Agreed